code/game/objects/structures/banners.dm

```diff
@@ -64,95 +79,279 @@
 
 /obj/structure/banner_frame/on_update_icon()
 	. = ..()
+
 	icon_state = base_icon_state
-	if(istype(banner))
-		var/image/I = image(banner.icon, "banner_base")
+	if(!istype(banner))
+		return
+
+	var/image/I = image(banner.icon, "[banner.icon_state]-hanging")
+	I.appearance_flags |= RESET_COLOR
+	I.color = banner.color
+	add_overlay(I)
+
+	if(banner.trim_color)
```

Comment:
kind of feel like the decals should be added to the banner state first, with `BLEND_INSET_OVERLAY`, and then we apply the trim on top to keep the layering sane. that way you could have user-added decals like stripes/crosses/etc that aren't mutually exclusive with other decals and work regardless of banner shape. this probably works fine as-is i just wanted to point out that inset overlay blend mode could be useful here now or in the future